### PR TITLE
React to corefxlab changes

### PR DESCRIPTION
**Pull request by davidfowl**
- Removed use of TryRead and follow the pin and use
pattern in the other APIs

### CODE REVIEW

File: NuGet.config [hunk 1]
```diff
@@ -1,6 +1,8 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <configuration>
   <packageSources>
+    <add key="CoreFxLabs" value="https://dotnet.myget.org/F/dotnet-corefxlab/api/v3/index.json" />
```
Comment (davidfowl) [inline NuGet.config]: This won't be merged. This branch just needs to exist for the mirror.

File: src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/HttpUtilities.cs [hunk 2]
```diff
@@ -270,23 +270,32 @@ internal unsafe static HttpVersion GetKnownVersion(byte* location, int length)
         /// <param name="knownScheme">A reference to the known scheme, if the input matches any</param>
         /// <returns>True when memory starts with known http or https schema</returns>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static bool GetKnownHttpScheme(this Span<byte> span, out HttpScheme knownScheme)
+        public static unsafe bool GetKnownHttpScheme(this Span<byte> span, out HttpScheme knownScheme)
         {
-            if (span.TryRead<ulong>(out var value))
+            fixed (byte* data = &span.DangerousGetPinnableReference())
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/HttpUtilities.cs]: Did they remove this API or is this a perf tweak?

Comment (davidfowl) [inline src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/HttpUtilities.cs]: It moved to another assembly and I didn't want to take a new dependency.

Comment (natemcmaster) [inline src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/HttpUtilities.cs]: ok, i don't know enough about what corefx is actually planning to release so this is fine for now.

**Pull request merged**